crates/subspace-farmer/src/single_disk_farm.rs

```diff
@@ -62,3 +66,66 @@ impl SingleDiskSemaphore {
         self.inner.access()
     }
 }
+
+/// An identifier for single plot farm, can be used for in logs, thread names, etc.
+#[derive(
+    Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize, From,
+)]
+pub struct SingleDiskFarmId(Ulid);
+
+impl fmt::Display for SingleDiskFarmId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+/// Metadata for `SingleDiskFarm`, stores important information about the contents of the farm
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum SingleDiskFarmMetadata {
+    /// V0 of the metadata
+    #[serde(rename_all = "camelCase")]
+    V0 {
+        /// ID of the farm
+        id: SingleDiskFarmId,
+        /// Genesis hash of the chain used for farm creation
+        #[serde(with = "hex::serde")]
+        genesis_hash: [u8; 32],
```

Comment:
Maybe better to use `Sha256Hash` here?

Reply:
No because it is not Sha256 hash, it is a genesis hash, which is Blake2b 256-bit.
I was considering to use `Vec<u8>`, but decided to go with fixed bytes array for now.